src/Media/Audio/OpenALSample16.cpp

```diff
@@ -42,8 +42,10 @@ bool AudioSample16::Open(PAudioDataSource data_source) {
         return false;
     }
 
-    alGenSources((ALuint)1, &al_source);
-    if (checkOpenALError()) {
+    alGenSources(1, &al_source);
+
+    // For some obscure reason we sometimes get al_source == -1 even though checkOpenALError() returns false. So we check it too.
+    if (checkOpenALError() || al_source == -1) {
```

Comment:
Is it worth logging something in the case al_source == -1?

Reply:
I honestly don't know what to log so that it actually helps us nail down the problem X).